server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/SieveScriptRoutes.java

```diff
@@ -0,0 +1,185 @@
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Empty script is not accepted"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "User not found")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = USER_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Username"),
+        @ApiImplicitParam(
+            name = SCRIPT_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Script name"),
+        @ApiImplicitParam(
+            required = false,
+            paramType = "query parameter",
+            dataType = "Boolean",
+            defaultValue = "False",
+            example = "?activate=true",
+            value = "If present, automatically activating the script.")
+    })
+    public void defineAddActiveSieveScript(Service service) {
+        service.put(USER_SCRIPT_PATH, this::addActiveSieveScript);
+    }
+
+    private HaltException addActiveSieveScript(Request request, Response response) throws UsersRepositoryException, QuotaExceededException, StorageException, ScriptNotFoundException {
+        User user = extractUser(request);
+        ScriptName script = extractScriptName(request);
+        boolean isActivated = isActivated(request.queryParams(ACTIVATE_PARAMS));
+        sieveRepository.putScript(user, script, extractSieveScriptFromRequest(request));
+        if (isActivated) {
+            sieveRepository.setActive(user, script);
+        }
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    private User extractUser(Request request) throws UsersRepositoryException {
+        String userName = Optional.ofNullable(request.params(USER_NAME))
+            .map(String::trim)
+            .filter(StringUtils::isNotEmpty)
+            .orElseThrow(() -> throw400withInvalidArgument("Invalid username"));
+
+        if (!usersRepository.contains(userName)) {
+            throw404("User not found");
+        }
+        return User.fromUsername(userName);
+    }
+
+    private ScriptName extractScriptName(Request request) {
+        return Optional.ofNullable(request.params(SCRIPT_NAME))
+            .map(String::trim)
+            .filter(StringUtils::isNotEmpty)
+            .map(ScriptName::new)
+            .orElseThrow(() -> throw400withInvalidArgument("Invalid Sieve script name"));
+    }
+
+    private ScriptContent extractSieveScriptFromRequest(Request request) {
+        return new ScriptContent(request.body());
+    }
+
+    private boolean isActivated(String activateParam) {
+        return Optional.of(Optional.ofNullable(activateParam)
+                .map(String::trim)
+                .orElse("false"))
```

Comment:
oh no, you are handling the default value at the wrong place and require to provide a hardcoded "false" as a string that will be parsed.

Reply:
I separated this method like this
```
private boolean isActivated(String activateParam) {
        return Optional.ofNullable(activateParam)
            .map(String::trim)
            .map(this::parseActivateParam)
            .orElse(false);
    }

    private boolean parseActivateParam(String activateParam) {
        if (activateParam.equalsIgnoreCase(Boolean.TRUE.toString())
            || activateParam.equalsIgnoreCase(Boolean.FALSE.toString())) {
            return Boolean.parseBoolean(activateParam);
        }

        throw throw400withInvalidArgument("Invalid activate query parameter");
    }
```